File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/repository/GoogleBooksRepository.java
package ru.kmorozov.library.data.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import ru.kmorozov.db.core.logic.model.book.base.BookInfo;

/**
 * Created by km on 21.12.2016.
 */
public interface GoogleBooksRepository extends MongoRepository<BookInfo, String> {

    BookInfo findByBookId(String bookId);
}

File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/repository/GoogleBooksRepository.kt
package ru.kmorozov.library.data.repository

import org.springframework.data.mongodb.repository.MongoRepository
import ru.kmorozov.db.core.logic.model.book.BookInfo

/**
 * Created by km on 21.12.2016.
 */
interface GoogleBooksRepository : MongoRepository<BookInfo, String> {

    fun findByBookId(bookId: String): BookInfo?
}
